torch/onnx/_internal/fx/serialization.py

```diff
@@ -135,23 +148,28 @@ def save_model_with_external_data(
             # Using torch.save wouldn't leverage mmap, leading to higher memory usage
             state_dict = el
         else:
-            try:
-                # Loads checkpoint using memory-map on CPU to support really large models
-                # The underlying torch.UntypedStorage is memory mapped, so state_dict is lazy loaded
-                state_dict = torch.load(el, map_location="cpu", mmap=True)
-            except (RuntimeError, ValueError) as e:
-                if "mmap can only be used with files saved with" in str(
-                    e
-                ) or isinstance(el, io.BytesIO):
-                    log.warning(
-                        "Failed to load the checkpoint with memory-map enabled, retrying without memory-map."
-                        "Consider updating the checkpoint with mmap by using torch.save() on PyTorch version >= 1.6."
-                    )
-                    if isinstance(el, io.BytesIO):
-                        el.seek(0)  # torch.load from `try:` has read the file.
-                    state_dict = torch.load(el, map_location="cpu")
-                else:
-                    raise e
+            if isinstance(el, str) and el.endswith(".safetensors"):
+                with tempfile.NamedTemporaryFile() as el_tmp:
+                    _convert_safetensors_to_torch_format(el, el_tmp.name)
```

Comment:
why (load from safetensor) -> (save as tmp torch.pt) -> (load from tmp torch.pt) -> state_dict,
but not (load from safetensor) -> state_dict?

Reply:
ouch, the reason is not in this pr by accident lol

by having a temorary file, we can mmap it and support large models even when the safetensor doesnt fit in memory. will update the pr

Reply:
But this step `(load from safetensor) -> (save as tmp torch.pt)` already requires fully loading weights into memory + more overhead of saving it to tmp file, or am I missing something?

Reply:
I will double check how the Storage differs when the tensor comes from disk or is created in memory.
I will change to safetensors -> state_dict to get unblocked in the mean time